### PR TITLE
Check that private link is not returned 

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What

Check that private link is not returned  unless download service token is given

### How to review

Review / test against https://github.com/ONSdigital/dp-filter-api/pull/78

### Who can review

Anyone

### CODE REVIEW

File: publishing/filterAPI/get_filter_output_test.go [hunk 1]
```diff
@@ -78,6 +79,39 @@ func TestSuccessfullyGetFilterOutput(t *testing.T) {
...
+				response.Value("links").Object().Value("filter_blueprint").Object().Value("href").String().Match("/filters/" + filterBlueprintID + "$")
+				response.Value("links").Object().Value("filter_blueprint").Object().Value("id").Equal(filterBlueprintID)
+				response.Value("links").Object().Value("self").Object().Value("href").String().Match("/filter-outputs/" + publishedFilterOutputID + "$")
+				response.Value("links").Object().Value("version").Object().Value("href").String().Match("/datasets/123/editions/2017/versions/1$")
+				response.Value("links").Object().Value("version").Object().Value("id").Equal("1")
+				response.Value("state").Equal("completed")
+			})
+		})
+
+		Convey("When making an authenticated requesting to get filter output with a valid download service token header", func() {
```
Comment (eldeal) [inline publishing/filterAPI/get_filter_output_test.go]: request not requesting :)

File: web/downloadService/get_public_filter_download_test.go [hunk 2]
```diff
@@ -0,0 +1 @@
+package downloadService
```
Comment (eldeal) [inline web/downloadService/get_public_filter_download_test.go]: why a file with only one line?

File: web/filterAPI/get_filter_output_test.go [hunk 3]
```diff
@@ -124,10 +125,46 @@ func TestSuccessfullyGetFilterOutput(t *testing.T) {
...
+				response.Value("links").Object().Value("filter_blueprint").Object().Value("href").String().Match("/filters/" + filterBlueprintID + "$")
+				response.Value("links").Object().Value("filter_blueprint").Object().Value("id").Equal(filterBlueprintID)
+				response.Value("links").Object().Value("self").Object().Value("href").String().Match("/filter-outputs/" + unpublishedFilterOutputID + "$")
+				response.Value("links").Object().Value("version").Object().Value("href").String().Match("/datasets/123/editions/2017/versions/1$")
+				response.Value("links").Object().Value("version").Object().Value("id").Equal("1")
+				response.Value("state").Equal("completed")
+			})
+		})
+
+		Convey("When making an request to get filter output with a download service token header", func() {
```
Comment (eldeal) [inline web/filterAPI/get_filter_output_test.go]: `a` request not `an` request

**Review by eldeal — COMMENTED** [review]
Generally looks fine to me, just a few typo/grammar things :)

File: publishing/filterAPI/get_filter_output_test.go (hunk 1 above)
Comment (CarlHembrough) [reply publishing/filterAPI/get_filter_output_test.go]: fixed 👍

File: web/downloadService/get_public_filter_download_test.go (hunk 2 above)
Comment (CarlHembrough) [reply web/downloadService/get_public_filter_download_test.go]: I committed these files by mistake - I have removed them from this branch 👍

File: web/filterAPI/get_filter_output_test.go (hunk 3 above)
Comment (CarlHembrough) [reply web/filterAPI/get_filter_output_test.go]: fixed

**Review by mattrout92 — APPROVED** [review]

**Pull request merged**